### ISSUE TITLE
[BUG] Chance-to-miss talents can't be countered by hit rating

### ISSUE THREAD

**Issue by Icedeadd** [ClassicWoWCommunity/cata-classic-bugs#516]
#### 1.1. Description
Dispel magic(rank 2) doesn't have more chance to dispel buffs(increased with talents by 30 % chance for dispel resist - Arcane subtlety, Silent resolve, Stoicism) with more hit rating.
  #### 1.2. How to Reproduce

           1. Equip more hit rating gear.
           2. Use dispel magic(rank 2) on Mage, priest, paladin (Arcane subtlety, Silent resolve, Stoicism)
           3. Use Weak auras for dispel resist.
           4. Do it for minimum 100 times.
  ###
# 1.3  Source Material
![Screenshot_31](https://user-images.githubusercontent.com/117484450/200162175-656e3074-c060-4a26-b481-f0526a2f2b48.png)

#### 2.1. Description
With more hit rating (4% +) it should decrease talents for dispel resist.
  #### 2.2 Source Material
![Screenshot_33](https://user-images.githubusercontent.com/117484450/200162262-b793be14-4351-4ec1-b44f-446587d45369.png)

**Comment by Evolvee** [comment]
When it comes to Dispel Magic(Rank 2), specifically, theres also some kind of weird "bug" that makes it only dispel 1 buff - maybe the chance to dispel is rolled separately - it should only dispel 0 or 2 buffs with the chance to dispel being rolled only once, not twice for each 1 cast, afaik.

**Comment by Icedeadd** [comment]
> When it comes to Dispel Magic(Rank 2), specifically, theres also some kind of weird "bug" that makes it only dispel 1 buff - maybe the chance to dispel is rolled separately - it should only dispel 0 or 2 buffs with the chance to dispel being rolled only once, not twice for each 1 cast, afaik.

Yeah, i m pretty sure that i watched all time to dispel only if 2 buffs are on person. (maybe missclick 2-3 times at worst). so there should be 200 dispels and i missing 22 of them. but yeah, i saw it happens a lot.

**Comment by JamminL** [comment]
> When it comes to Dispel Magic(Rank 2), specifically, theres also some kind of weird "bug" that makes it only dispel 1 buff - maybe the chance to dispel is rolled separately - it should only dispel 0 or 2 buffs with the chance to dispel being rolled only once, not twice for each 1 cast, afaik.

Dispel mechanic on priests seems to be working fine (as they were back then).
If you have 2 buffs;
- Rolls twice, 1 for each
If you only have 1 buff:
- It only rolls once
If you have only 1 buff and it has stacks:
Rolls once for 2 stacks (either removes 2 stacks or none)
If you have 2 buffs and 1 with stacks:
Rolls twice (could be 1 buff 1 stack, 2 stacks, 1 stack, 1 buff, 0)

Which is the behavior we researched back. Do you have any evidence this isn't true?

**Comment by Evolvee** [comment]
Nope, dont have any evidence, the description above matches this https://wowpedia.fandom.com/wiki/Dispel?oldid=1644346 https://github.com/Schaka/TBC-research/issues/26

**Comment by Icedeadd** [comment]
> 

Well i didn't even realised "this problem" before. But main idea behind bug post was that hit rating do not decrease talents for resist. As i wrote in post, blizzard said its fixed.

**Comment by XyzKangUI** [comment]
This should be reported as a PvE issue, because it is known those bugs get fixed within one week time.

https://www.bluetracker.gg/wow/topic/us-en/14287249236-new-dispel-resistance-mechanic/

That last sentence.

**Comment by Evolvee** [comment]
> This should be reported as a PvE issue, because it is known those bugs get fixed within one week time.
> 
> https://www.bluetracker.gg/wow/topic/us-en/14287249236-new-dispel-resistance-mechanic/
> 
> That last sentence mentions that only offensive spells dont benefit from talents that give resistance tho.. So I guess you have to fear/poly spam someone and try dispel that person and see if it resists in 1000 tries. I do agree that the blue post from your first screenshot is misleading as it also implies dispels. I guess that post should be read as non dispel spells cant be resisted by talents and racials when you have enough hit. E.g. You ice lance 1000x into a blood elf should not resist once with 6% hit or whatever hit cap + 2% is in wotlk.

❓ 
This post is about dispelling buffs such as stamina or intellect from unit protected by dispel resistance to these auras.
The post is not misleading in any way?

The "priest buff dispel resistance" is exactly the same as any other miss talent such as Hightened Senses or Sacred Cleansing. I dont get your post.

**Comment by Evolvee** [comment]
I didnt reply to you. I replied to xyzkangui who deleted his post aftet I posted mine.

**Comment by XyzKangUI** [comment]
> I didnt reply to you. I replied to xyzkangui who deleted his post after I posted mine.
> 
> Dunno why github doesnt display it in the log.

In fact you pmed me first on discord about my post. I cba'd to argue, then I deleted the post and you posted yours after.

**Comment by XyzKangUI** [comment]
> > > I didnt reply to you. I replied to xyzkangui who deleted his post after I posted mine.
> > > Dunno why github doesnt display it in the log.
> > 
> > 
> > In fact you pmed me first on discord about my post. I cba'd to argue, then I deleted the post and you posted yours after. Then you told me to repost.
> > @Icedeadd Sorry for the confusion. I linked a blue post mentioning that offensive spells do not benefit from the protection talents, while defensive spells such as stamina do. I questioned the fact that defensive spells by that same mechanism can be overcome by hit rating. From my point of view the talents such as Heigtened senses and blood elf racial only help you resist offensive spells done to you by others. I believe hit rating could even out those talents and racials to never miss your offensive spell. I just do not know if the same logic applies to the defensive spells. In classic 1.13.7 they added dispel resist and in tbc they made extra hit rating useless vs resistance talents, which included your offensive spells like fear having perma 15% resist chance vs priest with unbreakable will. Then the blue post from 2009 mentions a change that offensive spells do not benefit from spell resist talents, while defensive spells still do. Your first post implies as if spell hit should overcome both defensive and offensive spell, which I could not find any information about dating back to wotlk. Seeing from your test it seems that defensive spells indeed are not covered by the dispels. The blue post does mention that they changed the mechanic. Blizz is notorious for causing confusion. Like in their latest hotfix "immune to sap" is put under UI and accessibility changes. Furthermore they couldnt get cloak of shadows properly working entire tbc s1, so i wouldnt exclude the possibility of them messing up.
> > In this video you see the priest dispel and its removing 1/2 buffs a couple times, so I guess he had a resist. https://youtu.be/svJIdBWokZg&t=200
> 
> This issue is about the fact _(I didnt test it, just looking at the test results in the first image)_ that with this fix #143 they enabled auras that increase chance to miss to be penetrated by hit rating. Nothing more, nothing less. It doesnt matter whether the aura is offensive or deffensive.
> 
> Your bluepost from ghostcrawler merely explains what changes they made to the talents that increase such miss chance - before it was affecting all the spells, after the change it affected only the buffs/dots, not CCs for example. But the hit-rating-penetration mechanic should still apply to all spells, doesnt matter whether they are offensive or deffensive, its completely unrelated. If defensive spells are not covered, why would anyone even want to overcap hit rating for them? The change made by blizzard in #143 would be completely pointless.
> 
> The video you linked doesnt prove anything whatsoever as you simply cannot get 30% spell hit rating so yes, obviously some dispels on druid buffs will resist.
> 
> Dispel resistance is extremely crucial for PVP and whether this gets fixed or not has a huge impact on the gameplay. Lets try not to derail this thread from its purpose.

Good point about video. That is why i deleted it before anyone replied.

**Comment by Icedeadd** [comment]
> 

Yeah, I agree with both of you and have nothing important to add.  (Evolve wrote exactly everything that i wanted to say :D ).

**Comment by XyzKangUI** [comment]
> > 
> 
> Yeah, I agree with both of you and have nothing important to add. (In last post, Evolve wrote exactly everything that i wanted to say :D ).
![dispelresist](https://user-images.githubusercontent.com/94811434/200963710-70724ebd-5465-4f26-b6f3-273cc1799b68.png)

I have tested your post and come to the conclusion that the spell hit does not counter the talents at all.  Tried same with druid specced into Subtlety. Also it seems that there is no talent in game that actually covers offensive spells that do no damage, so that blue post by Ghostcrawler confused me. Good thing I deleted that entire post, but then again I got dragged back into this thread. All the dispel resistance talents changed in wotlk from "reduces chance your spells will be dispelled" to "helpful spells and DoTs"; dots clearly don't fall under the category of offensive spells doing no damage. 

Now the question remains if that blue post from your 1st post is not limited to only Heightened Senses, Arctic Wind, Blood elves' Magic Resistance. It mentions "spells to miss", are miss and resist the same? I believe they changed the combat text to avoid confusion, but in essence it is not one and the same. I will test vs a blood elf mage with / without arctic winds to see if any resists happen. Will edit this post after.

**Comment by XyzKangUI** [comment]
Now PTR is up again I done the same tests again.

1000 smites into blood elf = 0 resists. Meaning spell hit > racial.

![image](https://user-images.githubusercontent.com/94811434/203413254-5fd9dd3b-5b48-4d77-b657-a28e26d611fd.png)

There is a base dispel resistance in the game, which cannot be overcome by spell hit it seems. What that exact % is, I don't know.
![image](https://user-images.githubusercontent.com/94811434/203413367-3c53e7f4-7159-4c41-b3e6-785f67d987ac.png)

Also the talents like "Silent Resolve" and "Subtlety" cannot be overcome by spell hit.

This confirms my initial conclusion in the post I've deleted: "I guess that post should be read as non dispel spells cant be resisted by talents and racials when you have enough hit". 

Now the question is, is that the intended behaviour after that change from the blue post? I guess it is, but better to wait for an official response if one would take the time.

**Comment by Evolvee** [comment]
You have just repeated the first post. Dispel talents arent penetrated by hit rating...no shit Kevin, thats what this issue is all about... the dispel resist talent are exactly the same as global resist talents except only for certain auras but they get treated differently when it comes to hit rating penetration when they should not.

**Comment by Evolvee** [comment]
BUMP
can this be finally fixed? It goes directly against this https://github.com/JamminL/wotlk-classic-bugs/issues/143#issuecomment-1209998266

**Comment by JamminL** [comment]
Archiving old non-relevant issues in preparation for MOP. If I made a mistake please comment on it.
